packages/powersync_attachments_stream/lib/src/abstractions/sync_error_handler.dart

```diff
@@ -0,0 +1,36 @@
+import '../attachment.dart';
+
+/// Interface for handling errors during attachment operations.
+/// Implementations determine whether failed operations should be retried.
+/// Attachment records are archived if an operation fails and should not be retried.
+abstract class AbstractSyncErrorHandler {
```

Comment:
Let's make this an interface class and just name it `SyncErrorHandler`:

```suggestion
abstract interface class SyncErrorHandler {
```